File: JRC_Abogados.Server/Models/Caso.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JRC_Abogados.Server.Models
{
    public class Caso
    {
        [Key]
        public int Id { get; set; }
        public int TipoCasoId { get; set; }
        [ForeignKey("TipoCasoId")]
        public virtual TipoCaso? TipoCaso { get; set; }
        public int JuzgadoId { get; set; }
        [ForeignKey("JuzgadoId")]
        public virtual Juzgado? Juzgado { get; set; }
        public int UbicacionId { get; set; }
        [ForeignKey("UbicacionId")]
        public virtual Ubicacion? Ubicacion { get; set; }
        public string Descripcion { get; set; }
        public DateTime FechaInicio { get; set; }
        public DateTime? FechaTermino { get; set; }
        public int EstadoId { get; set; }
        [ForeignKey("EstadoId")]
        public virtual Estado? Estado { get; set; }
        public int ClienteId { get; set; }
        [ForeignKey("ClienteId")]
        public virtual Cliente? Cliente { get; set; }
        public int EmpleadoId { get; set; }
        [ForeignKey("EmpleadoId")]
        public virtual Empleado? Empleado { get; set; }
    }
}

File: jrc_abogados.client/src/app/Models/Caso.ts
import { Estado } from "./Estado";
import { Juzgado } from "./Juzgado";
import { TipoCaso } from "./TipoCaso";
import { Ubicacion } from "./Ubicacion";

export class Caso {
  id: number = 0;
  tipoCasoId = "";
  tipoCaso: TipoCaso = new TipoCaso;
  juzgadoId: number = 0;
  juzgado: Juzgado = new Juzgado;
  ubicacionId: number = 0;
  ubicacion: Ubicacion = new Ubicacion;
  descripcion: string = "";
  fechaInicio = "";
  fechaTermino = "";
  estadoId = "";
  estado: Estado = new Estado;
  empleadoId: number = 0;
  empleado: any;
  clienteId = "";
  cliente: any;
  [key: string]: any;
}
